index.d.ts

```diff
@@ -0,0 +1,209 @@
+): CancelablePromise<(EmittedType | EmittedTypeRest)[][]>;
+export function multiple<EmittedType>(
+	emitter: Emitter<EmittedType, any>,
+	event: string | symbol | (string | symbol)[],
+	options: MultipleOptions<EmittedType>
+): CancelablePromise<EmittedType[]>;
+
+/**
+ * Returns an [async iterator](http://2ality.com/2016/10/asynchronous-iteration.html) that lets you asynchronously
+ * iterate over events of `event` emitted from `emitter`. The iterator ends when `emitter` emits an event matching
+ * any of the events defined in `resolutionEvents`, or rejects if `emitter` emits any of the events defined in
+ * the `rejectionEvents` option.
+ */
+export function iterator<EmittedType, EmittedTypeRest = EmittedType>(
+	emitter: Emitter<EmittedType, EmittedTypeRest>,
+	event: string | symbol | (string | symbol)[],
+	options: IteratorMultiArgumentsOptions<EmittedType | EmittedTypeRest>
+): AsyncIterableIterator<(EmittedType | EmittedTypeRest)[]>;
+export function iterator<EmittedType>(
+	emitter: Emitter<EmittedType, any>,
+	event: string | symbol | (string | symbol)[],
+	filter: FilterFunction<EmittedType>
+): AsyncIterableIterator<EmittedType>;
+export function iterator<EmittedType>(
+	emitter: Emitter<EmittedType, any>,
+	event: string | symbol | (string | symbol)[],
+	options?: IteratorOptions<EmittedType>
+): AsyncIterableIterator<EmittedType>;
+
+export interface Options<EmittedType> {
+	/**
+	 * Events that will reject the promise.
+	 * @default ['error']
+	 */
+	rejectionEvents?: (string | symbol)[];
+
+	/**
+	 * By default, the promisified function will only return the first argument from the event callback,
+	 * which works fine for most APIs. This option can be useful for APIs that return multiple arguments
+	 * in the callback. Turning this on will make it return an array of all arguments from the callback,
+	 * instead of just the first argument. This also applies to rejections.
+	 *
+	 * @example
+	 *
+	 * const pEvent = require('p-event');
+	 * const emitter = require('./some-event-emitter');
+	 *
+	 * (async () => {
+	 *    const [foo, bar] = await pEvent(emitter, 'finish', {multiArgs: true});
+	 * })();
+	 *
+	 * @default false
+	 */
+	multiArgs?: boolean;
+
+	/**
+	 * Time in milliseconds before timing out.
+	 * @default Infinity
+	 */
+	timeout?: number;
+
+	/**
+	 * Filter function for accepting an event.
+	 *
+	 * @example
+	 *
+	 * const pEvent = require('p-event');
+	 * const emitter = require('./some-event-emitter');
+	 *
+	 * (async () => {
+	 *     const result = await pEvent(emitter, '🦄', value => value > 3);
+	 *     // Do something with first 🦄 event with a value greater than 3
+	 * })();
+	 */
+	filter?: FilterFunction<EmittedType>;
+}
+
+export interface MultiArgumentsOptions<EmittedType> extends Options<EmittedType> {
+	multiArgs: true;
+}
+
+export interface MultipleOptions<EmittedType> extends Options<EmittedType> {
+	/**
+	 * The number of times the event needs to be emitted before the promise resolves.
```

Comment:
Should document the default.

Reply:
There is no default for the `count` option, it is a mandatory option.

Reply:
`unknown` accepts anything too.

Reply:
Excuse me? This must be a typo...

Reply:
Yes, was on my phone on the go and commented on the wrong thread.